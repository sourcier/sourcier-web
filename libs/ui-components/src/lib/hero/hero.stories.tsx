import { Story, Meta } from '@storybook/react';

import { Hero, HeroProps } from './hero';
import { heading, copy } from './hero.mock';

export default {
  component: Hero,
  title: 'Hero',
} as Meta;

const Template: Story<HeroProps> = (args) => <Hero {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  heading,
  copy,
};
