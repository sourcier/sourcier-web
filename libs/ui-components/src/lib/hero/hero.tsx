export interface HeroProps {
  heading: string;
  copy: string;
}

export function Hero({ heading, copy }: HeroProps) {
  return (
    <div className="min-h-[75vh] hero">
      <div className="text-center hero-content">
        <div className="max-w-md">
          <h1 className="text-5xl font-bold">{heading}</h1>
          <p className="py-6">{copy}</p>
        </div>
      </div>
    </div>
  );
}

export default Hero;
