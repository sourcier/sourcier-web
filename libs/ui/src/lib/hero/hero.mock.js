export default {
  image: {
    src: 'https://source.unsplash.com/random',
    alt: 'Random image',
  },
  heading: 'Anim aute id magna',
  subHeading: 'aliqua ad deserunt',
  copy: 'Anim aute id magna aliqua ad ad non deserunt sunt. Qui irure qui lorem cupidatat commodo. Elit sunt amet fugiat',
};
