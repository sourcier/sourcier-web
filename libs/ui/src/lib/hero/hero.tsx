import React, { FC } from 'react';
export interface HeroProps {
  image: {
    src: string;
    alt: string;
  };
  heading: string;
  subHeading: string;
  copy: string;
}

export const Hero: FC<HeroProps> = ({ image, heading, subHeading, copy }) => {
  return (
    <main>
      <div>
        <div className="relative">
          <div className="absolute inset-x-0 bottom-0 bg-gray-100 h-1/2" />
          <div className="mx-auto max-w-7xl sm:px-6 lg:px-8">
            <div className="relative shadow-xl sm:rounded-2xl sm:overflow-hidden">
              <div className="absolute inset-0">
                <img
                  className="object-cover w-full h-full"
                  src={image.src}
                  alt={image.alt}
                  data-testid="hero-image"
                />
                <div className="absolute inset-0 bg-blue-700 mix-blend-multiply" />
              </div>
              <div className="relative px-4 py-16 sm:px-6 sm:py-24 lg:py-32 lg:px-8">
                <h1 className="text-4xl font-extrabold tracking-tight text-center sm:text-5xl lg:text-6xl">
                  <span className="block text-white" data-testid="hero-heading">
                    {heading}
                  </span>
                  <span
                    className="block text-indigo-200"
                    data-testid="hero-sub-heading"
                  >
                    {subHeading}
                  </span>
                </h1>
                <p
                  className="max-w-lg mx-auto mt-6 text-xl text-center text-indigo-200 sm:max-w-3xl"
                  data-testid="hero-copy"
                >
                  {copy}
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default Hero;
