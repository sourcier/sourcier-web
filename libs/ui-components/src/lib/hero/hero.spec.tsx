import renderer from 'react-test-renderer';

import Hero from './hero';
import { heading, copy } from './hero.mock';

describe('Hero', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(<Hero copy={copy} heading={heading} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
