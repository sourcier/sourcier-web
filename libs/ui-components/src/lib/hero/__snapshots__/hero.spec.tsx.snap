// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render successfully 1`] = `
<div
  className="min-h-[75vh] hero"
>
  <div
    className="text-center hero-content"
  >
    <div
      className="max-w-md"
    >
      <h1
        className="text-5xl font-bold"
      >
        Hello there
      </h1>
      <p
        className="py-6"
      >
        Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.
      </p>
    </div>
  </div>
</div>
`;
